assets/js/googlesitekit/data/create-snapshot-store.js

```diff
@@ -0,0 +1,206 @@
+		 */
+		*restoreSnapshot( { clearAfterRestore = true } = {} ) {
+			const { cacheHit, value } = yield {
+				payload: {},
+				type: RESTORE_SNAPSHOT,
+			};
+
+			if ( cacheHit ) {
+				yield {
+					payload: { snapshot: value },
+					type: SET_STATE_FROM_SNAPSHOT,
+				};
+
+				if ( clearAfterRestore ) {
+					yield {
+						payload: {},
+						type: DELETE_SNAPSHOT,
+					};
+				}
+			}
+
+			return cacheHit;
+		},
+
+		/**
+		 * Takes a snapshot of the current state of this store, using our cache API.
+		 *
+		 * @since n.e.x.t
+		 * @private
+		 *
+		 * @return {boolean} Cache write response.
+		 */
+		*takeSnapshot() {
+			const cacheResult = yield {
+				payload: {},
+				type: CREATE_SNAPSHOT,
+			};
+
+			return cacheResult;
+		},
+	};
+
+	const controls = {
+		[ DELETE_SNAPSHOT ]: () => {
+			return deleteItem( `datastore::cache::${ storeName }` );
+		},
+		[ CREATE_SNAPSHOT ]: createRegistryControl( ( registry ) => () => {
+			// console.log( `saving ${ storeName }`, registry.stores[ storeName ].store.getState() );
+			// registry.stores[ storeName ].store.getState().foo();
+			return setItem( `datastore::cache::${ storeName }`, registry.stores[ storeName ].store.getState() );
+		} ),
+		[ RESTORE_SNAPSHOT ]: () => {
+			// Only get snapshots made within the last hour.
+			return getItem( `datastore::cache::${ storeName }`, 3600 );
+		},
+	};
+
+	const reducer = ( state = INITIAL_STATE, { type, payload } ) => { // eslint-disable-line no-shadow
+		switch ( type ) {
+			case SET_STATE_FROM_SNAPSHOT: {
+				const { snapshot } = payload;
+				// Exclude any top-level errors from the restored state.
+				// eslint-disable-next-line no-unused-vars
+				const { error, ...newState } = snapshot;
+
+				return { ...newState };
+			}
+
+			default: {
+				return { ...state };
+			}
+		}
+	};
+
+	return { INITIAL_STATE, actions, controls, reducer };
+};
+
+/**
+ * Inspects a registry to find all stores that support our snapshot features.
+ *
+ * @since n.e.x.t
+ * @private
+ *
+ * @param {wp.data.registry} registry Optional. Registry object to inspect for stores that support state restoration. Defaults to `googlesitekit.data`.
+ * @return {Object} The snapshot store object.
+ */
+export const getStoresWithSnapshots = ( registry = Data ) => {
+	return Object.values( registry.namespaces ).filter( ( nameSpace ) => {
+		return Object.keys( nameSpace.getActions() ).includes( 'restoreSnapshot' );
+	} );
+};
+
+/**
+ * Takes snapshots of all supporting stores.
+ *
+ * @since n.e.x.t
+ * @private
+ *
+ * @param {wp.data.registry} [registry] Optional. Registry object to inspect for stores that support state restoration. Defaults to `googlesitekit.data`.
+ * @return {Promise} Promise resolves once all snapshots have been taken.
+ */
+export const snapshotAllStores = ( registry = Data ) => {
+	return Promise.all( getStoresWithSnapshots( registry ).map( ( store ) => {
+		return store.getActions().takeSnapshot();
```

Comment:
`store.getActions` returns _bound_ action creators which [returns the same thing as `registry.dispatch( STORE_NAME )`](https://github.com/WordPress/gutenberg/blob/4857ad58c1241b3d63d21a6880c989b85746c3dc/packages/data/src/registry.js#L153-L160) or `useDispatch( STORE_NAME )`.

You can see how they're created here:
https://github.com/WordPress/gutenberg/blob/4857ad58c1241b3d63d21a6880c989b85746c3dc/packages/data/src/namespace-store/index.js#L205-L211

In this case, `store` is actually `registry.stores[ STORE_NAME ]` rather than the "raw" definition.

So to answer your question, it wouldn't make sense to use dispatch here as the function will already do that. `dispatch` can only handle "raw" actions; these are [bound action creators](https://redux.js.org/api/bindactioncreators).